Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp

```diff
@@ -223,48 +223,48 @@ void WebFullScreenManager::enterFullScreenForElement(WebCore::Element* element,
-                return std::nullopt;
-
-            auto* image = renderImage->cachedImage()->image();
-            if (!image || !image->shouldUseQuickLookForFullscreen())
-                return std::nullopt;
-
-            auto* buffer = renderImage->cachedImage()->resourceBuffer();
-            if (!buffer)
-                return std::nullopt;
-
-            auto sharedMemoryBuffer = SharedMemory::copyBuffer(*buffer);
-            if (!sharedMemoryBuffer)
-                return std::nullopt;
-
-            return sharedMemoryBuffer->createHandle(SharedMemory::Protection::ReadOnly);
+    auto getMediaDetails = [&]() -> FullScreenMediaDetails {
+        CheckedPtr renderImage = dynamicDowncast<RenderImage>(element->renderer());
+        if (!renderImage)
+            return { };
+
+        auto* cachedImage = renderImage->cachedImage();
+        if (!cachedImage || cachedImage->errorOccurred())
+            return { };
+
+        auto* image = cachedImage->image();
+        if (!image || !image->shouldUseQuickLookForFullscreen())
+            return { };
+
+        auto* buffer = cachedImage->resourceBuffer();
+        if (!buffer)
+            return { };
+
+        auto imageSize = image->size();
+
+        auto mimeType = image->mimeType();
+        if (!MIMETypeRegistry::isSupportedImageMIMEType(mimeType))
+            mimeType = MIMETypeRegistry::mimeTypeForExtension(image->filenameExtension());
+        if (!MIMETypeRegistry::isSupportedImageMIMEType(mimeType))
+            mimeType = MIMETypeRegistry::mimeTypeForPath(cachedImage->url().string());
+        if (!MIMETypeRegistry::isSupportedImageMIMEType(mimeType))
+            return { };
+
+        auto sharedMemoryBuffer = SharedMemory::copyBuffer(*buffer);
+        if (!sharedMemoryBuffer)
+            return { };
+
+        auto imageResourceHandle = sharedMemoryBuffer->createHandle(SharedMemory::Protection::ReadOnly);
+
+        return {
+            FullScreenMediaDetails::Type::Image,
+            imageSize,
+            mimeType,
+            imageResourceHandle
         };
+    };
 
-        auto mimeType = emptyString();
-        if (auto* cachedImage = renderImage->cachedImage()) {
-            if (RefPtr image = cachedImage->image()) {
-                mimeType = image->mimeType();
-
-                if (!MIMETypeRegistry::isSupportedImageMIMEType(mimeType))
-                    mimeType = MIMETypeRegistry::mimeTypeForExtension(image->filenameExtension());
-            }
-            if (!MIMETypeRegistry::isSupportedImageMIMEType(mimeType))
-                mimeType = MIMETypeRegistry::mimeTypeForPath(cachedImage->url().string());
-        }
-
-        if (MIMETypeRegistry::isSupportedImageMIMEType(mimeType)) {
-            mediaDetails = {
-                FullScreenMediaDetails::Type::Image,
-                { },
-                mimeType,
-                getImageResourceHandle()
-            };
-        }
-    }
+    mediaDetails = getMediaDetails();
```

Comment:
it's a bit of a nit pick, but I would keep the `if (renderImage)` check outside of the lambda, and name it `getImageMediaDetails()`.
this way:
- we won't override any previously set mediaDetails when we don't have a RenderImage (if some code were to set it before this block in the future)
- it makes it clearer that there are other sources of media details like the code below